econml/dml.py

```diff
@@ -693,7 +736,303 @@ def transform(self, X):
+        `None`` means 1 unless in a :obj:`joblib.parallel_backend` context.
+        ``-1`` means using all processors. See :term:`Glossary <n_jobs>`
+        for more details.
+
+    verbose : int, optional (default=0)
+        Controls the verbosity when fitting and predicting.
+
+    random_state: int, :class:`~numpy.random.mtrand.RandomState` instance or None, optional (default=None)
+        If int, random_state is the seed used by the random number generator;
+        If :class:`~numpy.random.mtrand.RandomState` instance, random_state is the random number generator;
+        If None, the random number generator is the :class:`~numpy.random.mtrand.RandomState` instance used
+        by :mod:`np.random<numpy.random>`.
+    """
+
+    def __init__(self,
+                 model_y, model_t,
+                 discrete_treatment=False,
+                 n_crossfit_splits=2,
+                 n_estimators=100,
+                 criterion="mse",
+                 max_depth=None,
+                 min_samples_split=2,
+                 min_samples_leaf=1,
+                 min_weight_fraction_leaf=0.,
+                 max_features="auto",
+                 max_leaf_nodes=None,
+                 min_impurity_decrease=0.,
+                 subsample_fr='auto',
+                 honest=True,
+                 n_jobs=None,
+                 verbose=0,
+                 random_state=None):
+        model_final = SubsampledHonestForest(n_estimators=n_estimators,
+                                             criterion=criterion,
+                                             max_depth=max_depth,
+                                             min_samples_split=min_samples_split,
+                                             min_samples_leaf=min_samples_leaf,
+                                             min_weight_fraction_leaf=min_weight_fraction_leaf,
+                                             max_features=max_features,
+                                             max_leaf_nodes=max_leaf_nodes,
+                                             min_impurity_decrease=min_impurity_decrease,
+                                             subsample_fr=subsample_fr,
+                                             honest=honest,
+                                             n_jobs=n_jobs,
+                                             random_state=random_state,
+                                             verbose=verbose)
+        super().__init__(model_y=model_y, model_t=model_t,
+                         model_final=model_final, featurizer=None,
+                         discrete_treatment=discrete_treatment,
+                         n_splits=n_crossfit_splits, random_state=random_state)
+
+    def _get_inference_options(self):
+        # add statsmodels to parent's options
+        options = super()._get_inference_options()
+        options.update(blb=GenericSingleTreatmentModelFinalInference)
```

Comment:
The blb method currently takes no params. To do this we would need to create a child clas of GenericSingleTreatment, just for the sake of calling it here, but with no modification. I don't want to rename this particular class BLB inference, because it is more general. It holds for any time where model final supports predict_interval. Also in the docstring, I say that you can only use 'blb' not the class `GenericSingleTreatmentModelFinalInference`

Reply:
ok